src/main/java/com/shrralis/ssdemo1/controller/system/ExceptionHandlerControllerAdvice.java

```diff
@@ -12,38 +12,79 @@
 
 package com.shrralis.ssdemo1.controller.system;
 
+import com.shrralis.ssdemo1.exception.BadParameterFormatException;
 import com.shrralis.ssdemo1.exception.EntityNotUniqueException;
+import com.shrralis.tools.model.JsonError;
 import com.shrralis.tools.model.JsonResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MissingServletRequestParameterException;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * @author shrralis (https://t.me/Shrralis)
  * @version 1.0
  * Created 12/21/17 at 3:45 PM
  */
 @RestControllerAdvice
 public class ExceptionHandlerControllerAdvice {
+	private static final Logger logger = LoggerFactory.getLogger(ExceptionHandlerControllerAdvice.class);
+
     @ResponseStatus(value = HttpStatus.BAD_REQUEST)
     @ExceptionHandler(value = EntityNotUniqueException.class)
-    public JsonResponse entityNotUniqueHandler(EntityNotUniqueException e) {
-//        logger.error("EntityNotUniqueException: {}", e.getMessage());
-        return new JsonResponse(e.getMessage());
+    public JsonResponse entityNotUniqueExceptionHandler(EntityNotUniqueException e) {
+	    logger.error("EntityNotUniqueException: {}", e.getMessage());
+	    return new JsonResponse(e.getError());
+    }
+
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+	@ExceptionHandler(value = BadParameterFormatException.class)
+	public JsonResponse badParameterFormatExceptionHandler(BadParameterFormatException e) {
+		logger.error("BadParameterFormatException: {}", e.getMessage());
+		return new JsonResponse(e.getError());
     }
 
     @ResponseStatus(value = HttpStatus.BAD_REQUEST)
     @ExceptionHandler(value = IllegalArgumentException.class)
-    public String illegalArgumentException(IllegalArgumentException e) {
-//        logger.error("IllegalArgumentException: {}", e.getMessage());
-        return e.getMessage();
+    public JsonResponse illegalArgumentExceptionHandler(IllegalArgumentException e) {
+	    logger.error("IllegalArgumentException: {}", e.getMessage());
+
+	    Matcher matcher = Pattern.compile("\'([^\']*)\'").matcher(e.getMessage());
```

Comment:
Do you throw IllegalArgumentException? It might be better to use custom class with message field so we  do not need to parse exception